### PR TITLE
Closes #6 - Vulkano 0.28

### LINKED ISSUE #6: update to Vulkano 0.28, winit 0.26, egui 0.16
Hi, I've forked this repo for use in a personal project and managed to get the example running with Vulkano 0.28, winit 0.16, and egui 0.16. I figured I'd submit a PR just in case you or anyone else wanted to take a look.

### PR DESCRIPTION
**Pull request by VDorchester14**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -276,42 +291,41 @@ fn create_pipeline(
...
+        .vertex_input_state(BuffersDefinition::new().vertex::<Vertex>())
+        .vertex_shader(vs.entry_point("main").unwrap(), ())
+        .input_assembly_state(InputAssemblyState::new())
+        .viewport_state(ViewportState::viewport_dynamic_scissor_irrelevant())
+        .fragment_shader(fs.entry_point("main").unwrap(), ())
+        .rasterization_state(
+            RasterizationState::new()
+                .cull_mode(CullMode::None)
+        )
+        .color_blend_state(ColorBlendState::new(subpass.num_color_attachments()).blend_alpha())
```
Comment (derivator) [inline src/lib.rs:310]: Found the blending issue:

```suggestion
        .color_blend_state(ColorBlendState::new(subpass.num_color_attachments()).blend(blend))
```

File: src/lib.rs [hunk 2]
```diff
@@ -140,11 +140,14 @@ impl Painter {
...
-                .build()?,
-        );
+        let set = PersistentDescriptorSet::new(
+            layout.clone(),
+            [WriteDescriptorSet::image_view_sampler(
+                0,
+                ImageView::new(image)?,
+                self.sampler.clone()
+            )],
+        ).unwrap();
```
[the remark this answers is not in the document]
Comment (VDorchester14) [reply src/lib.rs]: I changed this one and the others handled with unwrap to use the ? operator. I think explicitly handling the error case and passing quitting is better than pushing that error down to some other point in the execution.

File: src/lib.rs [hunk 3]
```diff
@@ -209,15 +220,13 @@ impl Painter {
 
         let (vertex_buf, index_buf) = self.create_buffers((verts, indices))?;
         for (idx, clip) in clips.iter().enumerate() {
-            let mut ds = dynamic_state.clone();
             let mut scissors = Vec::with_capacity(1);
             let o = clip.min;
             let (w, h) = (clip.width() as u32, clip.height() as u32);
             scissors.push(Scissor {
-                origin: [o.x as i32, o.y as i32],
+                origin: [(o.x as i32).try_into().unwrap(), (o.y as i32).try_into().unwrap()],
```
[the remark this answers is not in the document]
Comment (VDorchester14) [reply src/lib.rs]: Also changed this one to your suggestion

File: src/lib.rs [hunk 4]
```diff
@@ -163,9 +165,18 @@ impl Painter {
         P: CommandPoolBuilderAlloc,
     {
         self.update_set(egui_ctx)?;
-        builder.next_subpass(Inline)?;
+        builder.next_subpass(Inline)
+            .unwrap()
```
[the remark this answers is not in the document]
Comment (VDorchester14) [reply src/lib.rs]: I changed this back to handling the error with the ? operator. I'm not sure what all there is to do beyond that

File: examples/example.rs [hunk 5]
```diff
@@ -424,15 +416,15 @@ impl Benchmark {
...
-            .hline(target);
+        Plot::new("plot")
+            .view_aspect(2.0)
+            .show(ui, |plot_ui| {
+                plot_ui.line(curve);
+                plot_ui.hline(zero);
+                plot_ui.hline(target);
+            });
 
         ui.label("Time in milliseconds that each frame took to draw:");
```
[the remark this answers is not in the document]
Comment (VDorchester14) [reply examples/example.rs:426]: Wasn't sure what you meant by this comment. It seems like the plotting in egui has been updated in the current version. I tried to mirror what was done in the [egui plot docs](https://docs.rs/egui/latest/egui/widgets/plot/struct.Plot.html) and call the proper plotting functions on the PlotUI struct in that closure. I can make changes tho if you think it needs them.

File: src/lib.rs (hunk 1 above)
Comment (VDorchester14) [reply src/lib.rs]: Good find, fixed this as well.

**Comment by VDorchester14** [comment]
Implemented all your suggestions except the plot one, because I wasn't quite sure what you meant. there's more comments on that conversation.

File: examples/example.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (derivator) [reply examples/example.rs:426]: Sorry, Github's interface didn't make this very understandable. I was referring to this line:

```rust 
ui.label("Time in milliseconds that each frame took to draw:");
```

This should happen before the plotting, so the label appears above the plot.

**Comment by derivator** [comment]
Fixes look good, thanks for the work! If you could fix the plot issue I will merge. I would also like if you could squash your 3 commits into a single commit for a clean history, but I can do that myself if needed.

File: examples/example.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (VDorchester14) [reply examples/example.rs]: Ah, ok yeah that makes sense. submitted that now.

**Comment by VDorchester14** [comment]
Just fixed the plotting issue and squashed commits into a single one. And no worries, it was great to contribute!

**Pull request merged**